lib/livebook/config.ex

```diff
@@ -39,27 +39,37 @@ defmodule Livebook.Config do
   """
   @spec file_systems() :: list(FileSystem.t())
   def file_systems() do
-    Application.fetch_env!(:livebook, :file_systems)
+    Application.fetch_env!(:livebook, :default_file_systems) ++
+      Enum.map(storage().all(:filesystem), &storage_to_fs/1)
   end
 
   @doc """
   Appends a new file system to the configured ones.
   """
   @spec append_file_system(FileSystem.t()) :: list(FileSystem.t())
-  def append_file_system(file_system) do
-    file_systems = Enum.uniq(file_systems() ++ [file_system])
-    Application.put_env(:livebook, :file_systems, file_systems, persistent: true)
-    file_systems
+  def append_file_system(%FileSystem.S3{} = file_system) do
```

Comment:
Calling it `append` is a bit misleading now because we don't have any order. Ideally, we would move these functions out of this module, because they are no longer related to config, and I would move all of them to `Livebook.FileSystem`, such as `Livebook.FileSystem.insert`, `Livebook.FileSystem.all`, and so on. But I think we can make this in a future pull request. What are your thoughts?

Reply:
I had exactly the same thoughts as yours, changing that in the next PR seems reasonable so that this one does not grow to significantly.

Reply:
I had another idea. We should probably have `Livebook.FileSystem.static()` (returns from config), `.dynamic()` (returns from storage), and `.all()` (both). Then on the settings page, we know that static cannot be added/deleted. But again, for another PR. :)